tests/devapps/MauiMacAppWithBroker/MainPage.xaml.cs

```diff
@@ -0,0 +1,119 @@
+﻿namespace MauiMacAppWithBroker;
+
+using Microsoft.Maui.Controls;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Broker;
+using System;
+using System.Runtime.InteropServices;
+
+public partial class MainPage : ContentPage
+{
+	private bool inUse = false;
+
+	public MainPage()
+	{
+		InitializeComponent();
+	}
+
+	private async void OnACIACSClicked(object sender, EventArgs e)
+	{
+		
+		if (inUse)
+		{
+			await Application.Current.MainPage.DisplayAlert("Warning", "Please wait until previous operation finished", "OK").ConfigureAwait(false);
+			return;
+		}
+		inUse = true;
+
+		SemanticScreenReader.Announce(CACIACSBtn.Text);
+
+        PublicClientApplicationBuilder builder = PublicClientApplicationBuilder
+			.Create("04b07795-8ddb-461a-bbee-02f9e1bf7b46")
+			.WithRedirectUri("msauth.com.msauth.unsignedapp://auth")
+			.WithClientCapabilities(new[] { "cp1" })
+			.WithAuthority("https://login.microsoftonline.com/common");
+		
+		builder = builder.WithLogging(SampleLogging);
+
+		builder = builder.WithBroker(new BrokerOptions(BrokerOptions.OperatingSystems.OSX)
+			{
+				ListOperatingSystemAccounts = false,
+				MsaPassthrough = false,
+				Title = "MSAL Dev App .NET FX"
+			}
+		);
+
+        IPublicClientApplication pca = builder.Build();
+
+        AcquireTokenInteractiveParameterBuilder interactiveBuilder = pca.AcquireTokenInteractive(new string[]{"https://graph.microsoft.com/.default"});
+
+		
+		AuthenticationResult result = await interactiveBuilder.ExecuteAsync(CancellationToken.None).ConfigureAwait(false);
+
+		IAccount account = result.Account;
+
+        AcquireTokenSilentParameterBuilder silentBuilder = pca.AcquireTokenSilent(new string[]{"https://graph.microsoft.com/.default"}, account);
+
+		result = await silentBuilder.ExecuteAsync(CancellationToken.None).ConfigureAwait(false);
+		
+
+		inUse = false;
+	}
+
+	private async void OnGetAllAccountsClicked(object sender, EventArgs e)
+	{
+		if (inUse)
+		{
+			await Application.Current.MainPage.DisplayAlert("Warning", "Please wait until previous operation finished", "OK").ConfigureAwait(false);
+			return;
+		}
+		inUse = true;
+
+		PublicClientApplicationBuilder builder = PublicClientApplicationBuilder
+			.Create("04b07795-8ddb-461a-bbee-02f9e1bf7b46")
```

Comment:
what app is this? can we not use one from the lab?

Reply:
I created an empty MAUI project; this id is generated automatically.

Reply:
Now I'm using a client id of my own app.